python/misc_utils/device.py

```diff
@@ -23,11 +23,23 @@
 
 import osmosdr
 import os
+from urllib.parse import parse_qsl
+from ast import literal_eval
 
 def get_devices(hint=""):
     return osmosdr.device_find(osmosdr.device_t(hint))
 
+def device_to_dict(dev):
+    dev_dict = {}
+    for k, v in parse_qsl(dev.to_string(), separator=","):
```

Comment:
If we want to avoid the import of `parse_qsl`, we can also do it this way:
```python3
def device_to_dict(dev):
    dev_dict = {}
    for pair in dev.split(","):
        k, v = pair.split("=")
        try:
            dev_dict[k] = literal_eval(v)
        except (ValueError, SyntaxError):
            dev_dict[k] = literal_eval(f"'{v}'")
    return dev_dict

```